### PR TITLE
Bug/parquet missing rows

### PR DESCRIPTION
**Pull request by Sajuno**
- Fixes a bug where rows were skipped in parquet file transformation
- Refactors file_test.go

### CODE REVIEW

File: runner/file_test.go [hunk 1]
```diff
@@ -570,32 +480,49 @@ c: d
...
 
-			rw.Close()
+			assert.Equal(t, out, exp)
+		})
+	}
+}
 
-			outTmpBs, err := ioutil.ReadFile(outTmp.Name())
-			assert.Nil(t, err)
+func transformTestFile[transformer func(string, *ResultWriter) error](filename string, transformFile transformer) (any, error) {
```
Comment (eatonphil) [inline runner/file_test.go:498]: Does this really need to be generic?

Comment (Sajuno) [reply runner/file_test.go:498]: It doesn't. Just started out that way :)

Comment (eatonphil) [reply runner/file_test.go]: Kk thanks!

**Review by eatonphil — APPROVED** [review]

**Comment by eatonphil** [comment]
Looks good!

**Pull request merged**